mantle/cmd/kola/testiso.go

```diff
@@ -394,11 +395,46 @@ func printSuccess(mode string) {
 	fmt.Printf("Successfully tested scenario %s for %s on %s (%s)\n", mode, kola.CosaBuild.Meta.OstreeVersion, kola.QEMUOptions.Firmware, metaltype)
 }
 
+func creatSSHAuthorizedKey() ([]byte, error) {
```

Comment:
Honoring the past, or a typo?
https://unix.stackexchange.com/questions/10893/what-did-ken-thompson-mean-when-he-said-id-spell-creat-with-an-e
:wink:

Reply:
I'd add a comment at the top like:
```
// createSSHAuthorizedKey generates just a public key to sanity check that Ignition accepts it (to reproduce a bug)
```

It would be nicer though if this was shared with `devshell.go` and also returned the private key (which devshell needs).
We could also change this code to actually ssh in to verify too.  But, that can wait for a larger refactoring.

Reply:
> Honoring the past, or a typo?
> https://unix.stackexchange.com/questions/10893/what-did-ken-thompson-mean-when-he-said-id-spell-creat-with-an-e
> wink

Oops, that was a typo. Let me go ahead and fix it.

Reply:
> I'd add a comment at the top like:
> 
> ```
> // createSSHAuthorizedKey generates just a public key to sanity check that Ignition accepts it (to reproduce a bug)
> ```
> 
> It would be nicer though if this was shared with `devshell.go` and also returned the private key (which devshell needs).
> We could also change this code to actually ssh in to verify too. But, that can wait for a larger refactoring.

I can create a follow-up PR to address this concern.

Reply:
> It would be nicer though if this was shared with `devshell.go` and also returned the private key (which devshell needs).

I was thinking about sharing the ssh key generation logic with `devshell.go`, but it looks like there are two variables in the `runDevShellSSH`function i.e. `tmpd`(https://github.com/coreos/coreos-assembler/blob/master/mantle/cmd/kola/devshell.go#L98), `sshKeyPath`(https://github.com/coreos/coreos-assembler/blob/master/mantle/cmd/kola/devshell.go#L104) that are explicitly used later in the function. So I decided not to change anything there and keep that logic as-is.

Reply:
How about this?
https://github.com/coreos/coreos-assembler/pull/1509/commits/d18379a70a82d588dfdc2631e34a41807e174007